packages/core/contracts/financial-templates/long-short-pair/LongShortPair.sol

```diff
@@ -269,66 +267,143 @@ contract LongShortPair is Testable, Lockable {
+     * @dev Will revert if: a) the contract is already early expire, b) it is before the expiration timestamp or c)
+     * an expire call has already been made.
+     */
+    function expire() public nonReentrant() notEarlyExpired() postExpiration() {
+        _requestOraclePrice(expirationTimestamp, customAncillaryData);
 
         emit ContractExpired(msg.sender);
     }
 
-    /****************************************
-     *      GLOBAL ACCESSORS FUNCTIONS      *
-     ****************************************/
+    /***********************************
+     *      GLOBAL VIEW FUNCTIONS      *
+     ***********************************/
+
     /**
      * @notice Returns the number of long and short tokens a sponsor wallet holds.
      * @param sponsor address of the sponsor to query.
-     * @return [uint256, uint256]. First is long tokens held by sponsor and second is short tokens held by sponsor.
+     * @return longTokens the number of long tokens held by the sponsor.
+     * @return shortTokens the number of short tokens held by the sponsor.
      */
-    function getPositionTokens(address sponsor) public view nonReentrantView() returns (uint256, uint256) {
+    function getPositionTokens(address sponsor)
+        public
+        view
+        nonReentrantView()
+        returns (uint256 longTokens, uint256 shortTokens)
+    {
         return (longToken.balanceOf(sponsor), shortToken.balanceOf(sponsor));
     }
 
+    /**
+     * @notice Generates a modified ancillary data that indicates the contract is being expired early.
+     */
+    function getEarlyExpirationAncillaryData() public view returns (bytes memory) {
+        return AncillaryData.appendKeyValueUint(customAncillaryData, "earlyExpiration", 1);
+    }
+
+    /**
+     * @notice Defines a special number that, if returned during an attempted early expiration, will cause the contract
+     * to do nothing and not expire. This enables the OO (and DVM voters in the case of a dispute) to choose to keep
+     * the contract running, thereby denying with the early settlement request.
+     */
+    function ignoreEarlyExpirationPrice() public pure returns (int256) {
+        return type(int256).min;
+    }
+
+    /**
+     * @notice If the earlyExpirationTimestamp is != 0 then a previous early expiration OO request might still be in the
+     * pending state. Check if the OO contains the ignore early price. If it does not contain this then the contract
+     * was early expired correctly. Note that _getOraclePrice call will revert if the price request is still pending,
+     * thereby reverting all upstream calls pre-settlement of the early expiration price request.
+     */
+    function isContractEarlyExpired() public returns (bool) {
+        return (earlyExpirationTimestamp != 0 &&
+            _getOraclePrice(earlyExpirationTimestamp, getEarlyExpirationAncillaryData()) !=
+            ignoreEarlyExpirationPrice());
```

Comment:
Does this mean that if someone triggers an invalid early expiration just before real expiration, they could delay real expiration for up to 4 days? I don't really have a suggestion, just double checking my thinking.

Reply:
yup! normal settlement is blocked until such time that the early expiration settlement has concluded. I had the unit test: `Can not call expire (normal expiration) if an early expiration is pending or correctly passed` which validated that it's blocked in the way you describe.

However, you are describing a subset of this test which is the expiration happens _just before_ warrants an additional test. I'm going to add one now to add additional colour on this to be 100% sure it behaves the way you described here.